internal/datastore/common/sql.go

```diff
@@ -562,17 +654,57 @@ func (tqs QueryExecutor) ExecuteQuery(
 		limit = *queryOpts.Limit
 	}
 
-	toExecute := query.limit(limit)
+	if limit < math.MaxInt64 {
+		query = query.limit(limit)
```

Comment:
Is this conditional just to reduce what's going over the wire?

Reply:
Basically, yes